br/pkg/storage/s3.go

```diff
@@ -64,10 +65,11 @@ const (
 	domainAliyun = "aliyuncs.com"
 )
 
-var permissionCheckFn = map[Permission]func(*s3.S3, *backuppb.S3) error{
-	AccessBuckets: checkS3Bucket,
-	ListObjects:   listObjects,
-	GetObject:     getObject,
+var permissionCheckFn = map[Permission]func(context.Context, s3iface.S3API, *backuppb.S3) error{
+	AccessBuckets:      checkS3Bucket,
+	ListObjects:        listObjects,
+	GetObject:          getObject,
+	PutAndDeleteObject: PutAndDeleteObjectCheck,
```

Comment:
naming style is slightly different 🤔  and seems no need to be public

Reply:
> seems no need to be public

the test is in a package named as `storage_test`